src/SFA.DAS.EAS.Portal/Database/DocumentClientFactory.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Extensions.Options;
+using SFA.DAS.EAS.Portal.Client.Configuration;
+
+namespace SFA.DAS.EAS.Portal.Database
+{
+    public class DocumentClientFactory : IDocumentClientFactory
+    {
+        private readonly IOptionsMonitor<CosmosDatabaseConfiguration> _options;
+
+        public DocumentClientFactory(IOptionsMonitor<CosmosDatabaseConfiguration> options)
```

Comment:
I think the recommendation is to not inject IOptions directly into your classes but to inject a configuration object tat has been set up by it.  Maybe raise some tech debt for this?

Reply:
Microsoft's [docs](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-2.2) seem to recommend injecting `IOptionsMonitor<>`.

Injecting `IOptionsMonitor<>` allows picking up config changes without restarting the host process, and at some point I'd like to enhance [AzureTableStorageConfigurationProvider](https://github.com/SkillsFundingAgency/das-shared-packages/blob/master/Configuration/SFA.DAS.Configuration.AzureTableStorage/AzureTableStorageConfigurationProvider.cs) to support that.